test/zkbob/ZkBobPool.t.sol

```diff
@@ -87,6 +91,70 @@ contract ZkBobPoolTest is Test {
         assertEq(bob.balanceOf(user3), 0.02 ether);
     }
 
+    function _setupNativeSwaps() internal {
+        vm.makePersistent(address(pool), address(bob));
+        vm.makePersistent(
+            EIP1967Proxy(payable(address(pool))).implementation(), EIP1967Proxy(payable(address(bob))).implementation()
+        );
+        vm.makePersistent(
+            address(pool.operatorManager()), address(pool.transfer_verifier()), address(pool.tree_verifier())
+        );
+
+        // fork mainnet
+        vm.createSelectFork(forkRpcUrl);
+
+        // create BOB-USDC 0.05% pool at Uniswap V3
+        deal(usdc, address(this), 1e12);
+        IERC20(usdc).approve(uniV3Positions, 1e12);
+        INonfungiblePositionManager(uniV3Positions).createAndInitializePoolIfNecessary(
+            usdc, address(bob), 500, TickMath.getSqrtRatioAtTick(276320)
+        );
+        INonfungiblePositionManager(uniV3Positions).mint(
+            INonfungiblePositionManager.MintParams({
+                token0: usdc,
+                token1: address(bob),
+                fee: 500,
+                tickLower: 276320,
+                tickUpper: 276330,
+                amount0Desired: 1e12,
+                amount1Desired: 0,
+                amount0Min: 0,
+                amount1Min: 0,
+                recipient: address(this),
+                deadline: block.timestamp
+            })
+        );
+
+        // enable token swaps for ETH
+        pool.setTokenSeller(address(new UniswapV3Seller(uniV3Router, address(bob), 500, usdc, 500)));
+    }
+
+    function testNativeWithdrawal() public {
```

Comment:
Can you add a test case when there is no enough liquidity in the USDC/BOB pool?

Reply:
Done. It is safe to try to sell more tokens than the available liquidity. In this case user will just receive all unsold BOB tokens back.